orbit/cmd/desktop/desktop.go

```diff
@@ -83,78 +48,89 @@ func main() {
 		myDeviceItem.Disable()
 		transparencyItem := systray.AddMenuItem("Transparency", "")
 
+		var insecureSkipVerify bool
+		if os.Getenv("FLEET_DESKTOP_INSECURE") != "" {
+			insecureSkipVerify = true
+		}
+
+		// TODO: figure out the right rootCA to pass to the client
+		client, err := service.NewDeviceClient(basePath, deviceToken, insecureSkipVerify, "")
+
+		if err != nil {
+			log.Printf("unable to initialize request client: %s", err)
+			os.Exit(1)
+		}
+
 		// Perform API test call to enable the "My device" item as soon
 		// as the device auth token is registered by Fleet.
-		go func() {
-			ticker := time.NewTicker(5 * time.Second)
-			defer ticker.Stop()
+		deviceEnabledChan := func() <-chan bool {
+			done := make(chan bool)
 
-			tr := http.DefaultTransport.(*http.Transport)
-			if os.Getenv("FLEET_DESKTOP_INSECURE") != "" {
-				// #nosec
-				tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
-			}
-			client := &http.Client{
-				Transport: tr,
-			}
+			go func() {
+				ticker := time.NewTicker(5 * time.Second)
+				defer ticker.Stop()
 
-			for {
-				resp, err := client.Get(devTestURL.String())
-				if err != nil {
-					// To ease troubleshooting we set the tooltip as the error.
-					myDeviceItem.SetTooltip(err.Error())
-					log.Printf("get device URL: %s", err)
-				} else {
-					resp.Body.Close()
-					if resp.StatusCode == http.StatusOK {
+				for {
+					_, err := client.ListDevicePolicies()
+
+					switch {
+					case err == nil:
 						myDeviceItem.SetTitle("My device")
 						myDeviceItem.Enable()
 						myDeviceItem.SetTooltip("")
+						done <- true
 						return
+					case errors.Is(err, service.ErrMissingLicense):
+						myDeviceItem.SetTitle("My device")
+						myDeviceItem.SetTooltip("")
+						done <- true
+						return
+					default:
+						// To ease troubleshooting we set the tooltip as the error.
+						myDeviceItem.SetTooltip(err.Error())
+						log.Printf("get device URL: %s", err)
 					}
+
+					<-ticker.C
 				}
-				<-ticker.C
-			}
+			}()
+
+			return done
 		}()
 
 		go func() {
+			<-deviceEnabledChan
 			tic := time.NewTicker(5 * time.Minute)
 			defer tic.Stop()
 
-			tr := http.DefaultTransport.(*http.Transport)
-			if os.Getenv("FLEET_DESKTOP_INSECURE") != "" {
-				// #nosec
-				tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
-			}
-			client := &http.Client{
-				Transport: tr,
-			}
-
 			for {
 				<-tic.C
-				// TODO: Use the policies endpoint instead of full device endpoint
-				// https://github.com/fleetdm/fleet/issues/5697
-				resp, err := client.Get(devTestURL.String())
-				if err != nil {
+
+				policies, err := client.ListDevicePolicies()
+				switch {
+				case err == nil:
+					// OK
+				case errors.Is(err, service.ErrMissingLicense):
+					myDeviceItem.SetTitle("My device")
+					myDeviceItem.Disable()
```

Comment:
We don't want to disable the button for free licenses.
For free licenses it just won't show any policy information in the tray or in the "device" page (no tab for policies).